### PR TITLE
[content-service] Fix BuildTarbal files owner

### PR DESCRIPTION
**Pull request by aledbf**
During the backup creation, an `IdentityMapping` helper is used to map the `UID` and `GID` from the host user namespace. When a backup is restored, the inverse process is executed.

This PR fixes an issue with the tar header fields `Uname` and `Gname` where the string is was preserved. We cannot assume such a thing exists outside the container.

### CODE REVIEW

File: components/content-service/pkg/archive/tar.go [hunk 1]
```diff
@@ -125,23 +124,23 @@ func ExtractTarbal(ctx context.Context, src io.Reader, dst string, opts ...TarOp
...
-			continue
-		}
-
-		err = remapFile(path.Join(dst, p), uid, gid)
-		if err != nil {
-			log.WithError(err).WithField("uid", uid).WithField("gid", gid).WithField("path", p).Warn("cannot chown")
+			err = remapFile(path.Join(dst, p), uid, gid)
+			if err != nil {
+				log.WithError(err).WithField("uid", uid).WithField("gid", gid).WithField("path", p).Warn("cannot chown")
+			}
```
Comment (rfay) [inline components/content-service/pkg/archive/tar.go:141]: I'm sure there are architectural reasons for this, but as you already know, the way to get a predictable tar unarchive is to extract with root privs on the system it's targeted for. That's far simpler than trying to maintain a mapping of uid/gid between two systems, which is almost certain to fail. But again, I know you're up against things that I have no idea about.

Comment (aledbf) [reply components/content-service/pkg/archive/tar.go]: @rfay the issue is not located in the restoration of the backup but in the creation. In particular, [this is the fix](https://github.com/gitpod-io/gitpod/pull/3636/files#diff-8cf81f14073bcad3e67f46c7667d82a56824f72ddffb9529e71944719e44a4afR163-R165)
If this works correctly I will refactor the `remapFile` helper

**Review by csweichel — APPROVED** [review]
LGTM

**Comment by shaal** [comment]
@aledbf thank you for working on this!
In order to test it, I need access to aledbf-remap.staging.gitpod-dev.com, but it seems down. 
(or if you can direct me how else I can test this?)

**Pull request merged**